src/useScrollSync.ts

```diff
@@ -0,0 +1,91 @@
+import type React from "react";
+import { useEffect } from "react";
+
+interface Options {
+  proportional?: boolean;
+  axis?: "horizontal" | "vertical" | "both";
+}
+
+export const useScrollSync = (
+  refs: React.MutableRefObject<HTMLElement | null>[],
+  { proportional = true, axis = "both" }: Options = {}
+): void => {
+  useEffect(() => {
+    const nodes = refs
+      .map((ref) => ref.current)
+      .filter((n) => n !== null) as HTMLElement[];
+    nodes.map((node) => {
+      node.onscroll = () =>
+        handleScroll(
+          node,
+          nodes.filter((n) => n !== node),
+          { proportional, axis }
+        );
+    });
+    return () => {
+      nodes.map((node) => {
+        node.onscroll = null;
+      });
+    };
+  }, []);
+};
+
+const handleScroll = (
+  leader: HTMLElement,
+  followers: HTMLElement[],
+  options: Options
+) => {
+  window.requestAnimationFrame(() => {
+    syncFollowers(leader, followers, options);
+  });
+};
+
+const syncFollowers = (
+  leader: HTMLElement,
+  followers: HTMLElement[],
+  options: Options
+) => {
+  followers.map((follower) => {
+    const { onscroll } = follower;
+    follower.onscroll = null;
+    syncFollower(leader, follower, options);
+    window.requestAnimationFrame(() => {
+      follower.onscroll = onscroll;
+    });
+  });
+};
+
+const syncFollower = (
+  leader: HTMLElement,
+  follower: HTMLElement,
+  options: Options
+) => {
+  const {
+    scrollTop,
+    scrollHeight,
+    clientHeight,
+    scrollLeft,
+    scrollWidth,
+    clientWidth,
+  } = leader;
+
+  const scrollTopOffset = scrollHeight - clientHeight;
+  const scrollLeftOffset = scrollWidth - clientWidth;
+
+  const { proportional, axis } = options;
+
+  /* Calculate the actual pane height */
+  const paneHeight = follower.scrollHeight - clientHeight;
+  const paneWidth = follower.scrollWidth - clientWidth;
+  /* Adjust the scrollTop position of it accordingly */
+  if ((axis === "both" || axis === "vertical") && scrollTopOffset > 0) {
+    follower.scrollTop = proportional
+      ? (paneHeight * scrollTop) / scrollTopOffset
+      : scrollTop; // eslint-disable-line
+  }
+  if ((axis === "both" || axis === "horizontal") && scrollLeftOffset > 0) {
+    follower.scrollLeft = proportional
+      ? (paneWidth * scrollLeft) / scrollLeftOffset
+      : scrollLeft; // eslint-disable-line
```

Comment:
Why is linter has to be disabled?

Reply:
Not sure actually. I copied most of this from the [existing implementation](https://github.com/okonet/react-scroll-sync/blob/master/src/ScrollSync.js#L119). I tried removing the `eslint-disable-line`, but my editor keeps adding it back in... I'll investigate to see if it's just my editor config or something else.

Reply:
🤷 not sure why the linter was being disabled before. I removed the `eslint-disable-line` comment and things seem to do just fine. so this should now be addressed.